src/figuresClasses.ts

```diff
@@ -1,11 +1,96 @@
-export interface Figure {}
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: 'triangle';
+
+  public color: Color;
+
+  public a: number;
+
+  public b: number;
+
+  public c: number;
+
+  constructor(color: Color, a: number, b: number, c: number) {
+    this.shape = 'triangle';
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
```

Comment:
```suggestion
  constructor(public color: Color, public a: number, public b: number, public c: number) {
```
If you add a modifier directly in the constructor you don't have to assign values manually, typescript will do it for you